### PR TITLE
fix: [BT-5900]: dependency breakdown

### PR DESCRIPTION
**Pull request by VivekHarness**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56613)

### CODE REVIEW

**Comment by VivekHarness** [comment]
trigger messagecheck

**Comment by VivekHarness** [comment]
trigger uts

**Comment by VivekHarness** [comment]
trigger uts

**Comment by VivekHarness** [comment]
trigger uts

**Comment by VivekHarness** [comment]
trigger ut5

**Comment by VivekHarness** [comment]
trigger ut1

**Comment by VivekHarness** [comment]
trigger ut5

**Comment by VivekHarness** [comment]
trigger ut1

**Comment by VivekHarness** [comment]
trigger uts

**Comment by VivekHarness** [comment]
trigger ut1

**Comment by VivekHarness** [comment]
trigger ut9

**Comment by VivekHarness** [comment]
trigger ut1

**Comment by VivekHarness** [comment]
trigger ut9

**Comment by VivekHarness** [comment]
trigger ut9

**Review by AnshuH — APPROVED** [review]

File: idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel [hunk 1]
```diff
@@ -13,6 +13,7 @@ java_library(
     deps = [
         "//:lombok",
         "//953-events-api:module",
+        "//980-commons/src/main/java/software/wings/utils:module",
```
Comment (vikyathharekal) [inline idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel:16]: This module uses classes from "//953-events-api:module" module only.
Why are we adding "//980-commons/src/main/java/software/wings/utils:module"? Do you see anything broken here?

Comment (VivekHarness) [reply idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel:16]: Yeah I got compile check issue when this dependency is not added. There was a class that shows missing after these changes, so I have to add this dependency to remove the error.

Comment (vikyathharekal) [reply idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel:16]: Ok, it is working for us locally and in the build pipelines. Not sure how without this change

**Review by vikyathharekal — APPROVED** [review]

File: idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel (hunk 1 above)
Comment (VivekHarness) [reply idp-service/src/main/java/io/harness/idp/events/producers/BUILD.bazel:16]: May be cause of being specific with the dependency, I was getting this error. But overall the build time is being reduced by quite a lot.

**Pull request merged**